sdcm/nemesis.py

```diff
@@ -1088,6 +1089,70 @@ def disrupt_ldap_connection_toggle(self):
         ContainerManager.unpause_container(self.tester.localhost, 'ldap')
         self.log.info('finished with nemesis')
 
+    def disrupt_ldap_role_add_remove_permission(self):
+        """
+        Scenario:
+        1. Create new user in Scylla
+        2. Create new role in Ldap containing the new user as a member.
+        3. Verify this new user is unauthorised on resources (create a keyspace).
+        4. Create a new super-user role in Scylla.
+        5. Create a new super-user role in Ldap, with the new user as a member.
+        6. Verify this new user is authorised on resources.
+        7. Modify Ldap role to remove membership of new user.
+        8. Verify the new user is unauthorised on resources (create a table).
+        9. Do clean-up: delete Ldap Role, Scylla roles and keyspace/tables.
+
+        """
+        if not self.cluster.params.get('use_ldap_authorization'):
+            raise UnsupportedNemesis('Cluster is not configured to run with LDAP authorization, hence skipping')
+        if not self.cluster.params.get('ldap_server_type') == LdapServerType.MS_AD:
```

Comment:
this should be added only after proving this nemesis won't work with MS AD, not because it did not work for you

Reply:
testing..

Reply:
After updating with the correct Ldap-server address for the case of MS_AD, looks like it fails to work.
Got a failure for the following code (using MS_AD + saslauth):
```
        self.tester.create_role_in_scylla(node=node, role_name=new_test_user, is_superuser=False,
                                          is_login=True)
        if self.cluster.params.get('prepare_saslauthd'):
            self.tester.add_user_in_ldap(username=new_test_user)
        with pytest.raises(Unauthorized, match="has no CREATE permission"):
            with self.cluster.cql_connection_patient(node=node, user=new_test_user, password=LDAP_PASSWORD) as session:
                session.execute(
                    """ CREATE KEYSPACE IF NOT EXISTS customer_ldap WITH replication = {'class': 'SimpleStrategy',
                    'replication_factor': 1} """)
```
It failed for "incorrect credentials":
```
2023-01-23 08:45:49.381: (DisruptionEvent Severity.ERROR) period_type=end event_id=4f3e6a0c-f3d5-44e4-9331-cf5ed8280e9d duration=2m15s: nemesis_name=LdapRoleAddRemovePermission target_node=Node test-ms-ad-master-test-ms--db-node-bce8222f-2 [13.48.26.227 | 10.0.1.239] (seed: False) errors=('Unable to connect to any servers', {'10.0.0.90:9042': AuthenticationFailed('Failed to authenticate to 10.0.0.90:9042: Error from server: code=0100 [Bad credentials] message="Incorrect credentials"'), '10.0.1.239:9042': AuthenticationFailed('Failed to authenticate to 10.0.1.239:9042: Error from server: code=0100 [Bad credentials] message="Incorrect credentials"')})
Traceback (most recent call last):
File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 4104, in wrapper
result = method(*args[1:], **kwargs)
File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1141, in disrupt_ldap_role_add_remove_permission
with self.cluster.cql_connection_patient(node=node, user=new_test_user, password=LDAP_PASSWORD) as session:
File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/decorators.py", line 67, in inner
return func(*args, **kwargs)
File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 3447, in cql_connection_patient
return self.cql_connection(**kwargs)
File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 3403, in cql_connection
return self._create_session(node=node, keyspace=keyspace, user=user, password=password,
File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 3380, in _create_session
session = cluster_driver.connect()
File "cassandra/cluster.py", line 1733, in cassandra.cluster.Cluster.connect
File "cassandra/cluster.py", line 1769, in cassandra.cluster.Cluster.connect
File "cassandra/cluster.py", line 1756, in cassandra.cluster.Cluster.connect
File "cassandra/cluster.py", line 3572, in cassandra.cluster.ControlConnection.connect
File "cassandra/cluster.py", line 3617, in cassandra.cluster.ControlConnection._reconnect_internal
cassandra.cluster.NoHostAvailable: ('Unable to connect to any servers', {'10.0.0.90:9042': AuthenticationFailed('Failed to authenticate to 10.0.0.90:9042: Error from server: code=0100 [Bad credentials] message="Incorrect credentials"'), '10.0.1.239:9042': AuthenticationFailed('Failed to authenticate to 10.0.1.239:9042: Error from server: code=0100 [Bad credentials] message="Incorrect credentials"')})
```

Reply:
Why?

Reply:
because the user has no permissions to modify the MS-AD server.